setup.json

```diff
@@ -19,105 +19,87 @@
-    "django==1.11.25",
-    "ete3==3.1.1",
-    "graphviz==0.13",
+    "aldjemy~=0.9.1",
+    "alembic~=1.2",
+    "circus~=0.15.0",
+    "click-completion~=0.5.1",
+    "click-config-file~=0.5.0",
+    "click-spinner~=0.1.8",
+    "click~=7.0",
+    "django~=2.2",
+    "ete3~=3.1",
+    "graphviz~=0.13",
     "ipython>=4.0,<6.0",
-    "kiwipy[rmq]==0.5.1",
-    "mock==3.0.5",
-    "numpy==1.16.4",
-    "paramiko==2.6.0",
-    "passlib==1.7.1",
-    "pika==1.1.0",
-    "plumpy==0.14.3",
-    "psutil==5.6.3",
-    "psycopg2-binary==2.8.3",
-    "pyblake2==1.1.2; python_version<'3.6'",
-    "python-dateutil==2.8.0",
-    "pytz==2019.3",
-    "pyyaml==3.13",
-    "reentry>=1.3.0",
-    "simplejson==3.16.0",
-    "sqlalchemy-utils==0.34.2",
-    "sqlalchemy==1.3.10",
-    "tabulate==0.8.5",
+    "jinja2~=2.10",
+    "kiwipy[rmq]~=0.5.1",
+    "numpy~=1.17",
+    "paramiko~=2.6",
+    "pika~=1.1",
+    "plumpy~=0.14.3",
+    "psutil~=5.6",
+    "psycopg2-binary~=2.8",
+    "pyblake2~=1.1; python_version<'3.6'",
+    "python-dateutil~=2.8",
+    "pytz~=2019.3",
+    "pyyaml~=3.13",
+    "reentry~=1.3",
+    "simplejson~=3.16",
+    "sqlalchemy-utils~=0.34.2",
+    "sqlalchemy~=1.3",
+    "tabulate~=0.8.5",
     "tornado<5.0",
-    "tzlocal==2.0.0",
-    "upf_to_json==0.9.2",
-    "uritools==2.2.0",
-    "wrapt==1.11.2"
+    "tzlocal~=2.0",
+    "upf_to_json~=0.9.2",
+    "wrapt~=1.11"
   ],
   "extras_require": {
     "ssh_kerberos": [
-      "gssapi==1.6.1",
-      "pyasn1==0.4.7"
+      "gssapi~=1.6",
+      "pyasn1~=0.4.8"
     ],
     "rest": [
-      "flask-cache==0.13.1",
-      "flask-cors==3.0.8",
-      "flask-httpauth==3.3.0",
-      "flask-marshmallow==0.10.1",
-      "flask-restful==0.3.7",
-      "flask-sqlalchemy==2.4.1",
-      "flask==1.1.1",
-      "itsdangerous==1.1.0",
-      "marshmallow-sqlalchemy==0.19.0",
-      "pyparsing==2.4.2",
-      "python-memcached==1.59",
-      "seekpath==1.9.3",
-      "sqlalchemy-migrate==0.12.0"
+      "flask-cors~=3.0",
+      "flask-restful~=0.3.7",
+      "flask~=1.1",
+      "pyparsing~=2.4",
+      "python-memcached~=1.59",
+      "seekpath~=1.9"
     ],
     "docs": [
       "docutils==0.15.2",
-      "jinja2==2.10.3",
-      "markupsafe==1.1.1",
-      "pygments==2.4.2",
-      "sphinx-rtd-theme==0.4.3",
-      "sphinx==2.2.0",
-      "sphinxcontrib-contentui==0.2.4",
-      "sphinxcontrib-details-directive==0.1.0"
+      "pygments~=2.5",
+      "sphinx-rtd-theme~=0.4.3",
+      "sphinx~=2.2",
+      "sphinxcontrib-contentui~=0.2.4",
+      "sphinxcontrib-details-directive~=0.1.0"
     ],
     "atomic_tools": [
-      "PyCifRW==4.4.1",
-      "ase==3.17.0",
-      "monty==2.0.4",
-      "pymatgen<=2018.12.12",
-      "pymysql==0.9.3",
-      "seekpath==1.9.3",
-      "spglib==1.14.1.post0"
+      "PyCifRW~=4.4",
+      "ase~=3.18",
+      "pymatgen>=2019.7.2",
+      "pymysql~=0.9.3",
+      "seekpath~=1.9",
+      "spglib~=1.14"
     ],
     "notebook": [
       "jupyter==1.0.0",
```

Comment:
I think these requirements were to avoid collisions with py2 and maybe now we can just put `jupyter` and `notebook`, without version specification?

Reply:
No unfortunately we can't. It's true that `notebook>=6` does not support python 2, but it also requires `pyzmq>=17` and `tornado>=5` and we, because of `circus` (and `plumpy` and `kiwipy`) have exactly `pyzmq<17` and `tornado<5`. I had made a start into seeing whether I could fix circus and make it compatible with the newer versions and made some headway but it is not trivial. Hopefully I can look into this more with @muhrin during the coding week